### PR TITLE
Add clang tidy and CI

### PR DESCRIPTION
**Pull request by externl**
Add .clang-tidy file with a few configured checks.

### CODE REVIEW

File: cpp/Ice/optional/ContactDBI.cpp [hunk 1]
```diff
@@ -21,15 +21,15 @@ ContactDBI::addContact(
     contact->name = name;
     if (type)
     {
-        contact->type = std::move(type);
+        contact->type = type;
     }
     if (number)
     {
-        contact->number = std::move(number);
+        contact->number = number;
```
Comment (pepone) [inline cpp/Ice/optional/ContactDBI.cpp:28]: What was the problem with this `std::move` call?

**Review by pepone — APPROVED** [review]

File: cpp/Ice/optional/ContactDBI.cpp (hunk 1 above)
Comment (externl) [reply cpp/Ice/optional/ContactDBI.cpp:28]: This type is trivially copyable. Using `std::move` doesn't do anything.

Comment (pepone) [reply cpp/Ice/optional/ContactDBI.cpp:28]: I guess we want to do `std::move(*number)` to move the string?

Comment (externl) [reply cpp/Ice/optional/ContactDBI.cpp]: Removing for number was a mistake. I've added it back.

**Pull request merged**